Sources/StreamChatUI/VoiceRecording/Components/AudioQueuePlayerNextItemProvider.swift

```diff
@@ -0,0 +1,154 @@
+        }
+
+        let nextVoiceRecordingInCurrentMessage = findVoiceRecordingAttachmentAfter(
+            attachmentWithURL: currentVoiceRecordingURL,
+            in: currentVoiceRecordingMessage
+        )
+
+        switch lookUpScope {
+        case .sameMessage:
+            return nextVoiceRecordingInCurrentMessage?.voiceRecordingURL
+
+        case .subsequentMessagesFromUser:
+            if let nextVoiceRecordingInCurrentMessage = nextVoiceRecordingInCurrentMessage {
+                return nextVoiceRecordingInCurrentMessage.voiceRecordingURL
+
+            } else if let nextMessage = findMessageBefore(
+                message: currentVoiceRecordingMessage,
+                in: messages
+            ), nextMessage.author.id == currentVoiceRecordingMessage.author.id {
+                return nextMessage.voiceRecordingAttachments.first?.voiceRecordingURL
+
+            } else {
+                return nil
+            }
+
+        default:
+            return nil
+        }
+    }
+
+    // MARK: - Helpers
+
+    private func voiceRecordingMessages(
+        in messages: [ChatMessage],
+        includeMessagesSentByCurrentUser: Bool
+    ) -> [ChatMessage] {
+        messages.filter { message in
+            guard
+                !message.isSentByCurrentUser || includeMessagesSentByCurrentUser,
+                !message.voiceRecordingAttachments.isEmpty
+            else {
+                return false
+            }
+            return true
+        }
+    }
+
+    private func findVoiceRecordingMessage(
+        in messages: [ChatMessage],
+        containingAttachmentWithURL url: URL
+    ) -> ChatMessage? {
+        messages.first { message in
+            message.voiceRecordingAttachments.first { attachment in
+                attachment.voiceRecordingURL == url
+            } != nil
+        }
+    }
+
+    private func findMessageBefore(
+        message: ChatMessage,
+        in messages: [ChatMessage]
+    ) -> ChatMessage? {
+        guard
+            let messageIndex = messages.lastIndex(of: message)
+        else {
+            return nil
+        }
+
+        let indexBefore = messages.index(before: messageIndex)
+
+        guard messages.indices.contains(indexBefore) else {
+            return nil
+        }
+
+        return messages[indexBefore]
+    }
+
+    private func findVoiceRecordingAttachmentAfter(
+        attachmentWithURL url: URL,
+        in message: ChatMessage
+    ) -> ChatMessageVoiceRecordingAttachment? {
+        let voiceRecordingAttachments = message.voiceRecordingAttachments
+        guard
+            let attachment = voiceRecordingAttachments.first(where: { $0.voiceRecordingURL == url }),
+            let attachmentIndex = voiceRecordingAttachments.firstIndex(of: attachment)
+        else {
+            return nil
+        }
+
+        let indexAfter = voiceRecordingAttachments.index(after: attachmentIndex)
+
+        guard voiceRecordingAttachments.indices.contains(indexAfter) else {
```

Comment:
In StreamChatUI you have an extension over Collection that allows you to do `voiceRecordingAttachments[safe: indexAfter]` and will return an optional value that you can guard let against

Reply:
We should probably move that to StreamChat, though

Reply:
For now i'm going to change to use the extension. Thank you :)